internal/app/storage/storage.go

```diff
@@ -92,3 +94,42 @@ func (strg *Storage) RestoreFromFile() {
 		logger.Initialize().Info("Ошибка при unmarshal ", err)
 	}
 }
+
+func (strg *Storage) RestoreDB() {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+
+	_, err := strg.DB.ExecContext(ctx, `
+	CREATE TABLE IF NOT EXISTS urls (
+	    "correlation_id" TEXT,
+		"longurl" TEXT UNIQUE,
+		"shorturl" TEXT
+	)
+	`)
+	if err != nil {
+		logger.Initialize().Info("Ошибка при создании таблицы urls ", err)
+	}
+
+}
+
+func (strg *Storage) SaveToDB(longurl, shorturl string) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+
+	query := fmt.Sprintf("INSERT INTO urls (longurl, shorturl) VALUES ('%s', '%s')", longurl, shorturl)
+	_, err := strg.DB.ExecContext(ctx, query)
+	if err != nil {
+		var pgErr *pgconn.PgError
+		if errors.As(err, &pgErr) && pgerrcode.IsIntegrityConstraintViolation(pgErr.Code) {
+			query = fmt.Sprintf("SELECT shorturl FROM urls WHERE longurl = '%s'", longurl)
+			ans := strg.DB.QueryRowContext(ctx, query)
+			var shorturl2 string
+			err2 := ans.Scan(&shorturl2)
```

Comment:
```suggestion
			err = ans.Scan(&shorturl)
```
нет смысла создавать новую переменную. Ну и в целом имя не очень удачное

Ошибку мы тоже тут обработали и можно спокойно ее заменить на новую